modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt

```diff
@@ -163,33 +170,106 @@ internal class DiscoverAdapter(
     }
 
     inner class CarouselListViewHolder(var binding: RowCarouselListBinding) : NetworkLoadableViewHolder(binding.root) {
+        private var autoScrollHelper: AutoScrollHelper? = null
+        private val scrollListener = object : OnScrollListener() {
+            private var draggingStarted: Boolean = false
+
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                when (newState) {
+                    RecyclerView.SCROLL_STATE_SETTLING -> Unit // Do nothing
+                    RecyclerView.SCROLL_STATE_DRAGGING -> {
+                        draggingStarted = true
+                        autoScrollHelper?.stopAutoScrollTimer()
+                    }
+                    RecyclerView.SCROLL_STATE_IDLE -> {
+                        if (draggingStarted) {
+                            /* Start auto scroll with a delay after a manual swipe */
+                            autoScrollHelper?.startAutoScrollTimer(delay = AutoScrollHelper.AUTO_SCROLL_DELAY)
```

Comment:
Personally, I would make the additional delay after a manual swipe something much higher like 5 seconds (or even higher) in order to make it very unlikely that the user would still be looking at the carousel item when the auto scroll picked up again. But that's just my personal preference, I'll defer to @adamzelinski on this.

Reply:
The manual swipe delay doesn't look as bad to me as the auto-scroll. Although it's hard to tell when you're not looking at it on your device. I think if we update the auto and manual resume scroll to 5 seconds this might work for both.